### ISSUE TITLE
Not able to display list of blocks in CKEditor

### ISSUE THREAD

**Issue by stpaultim** [backdrop-contrib/ckeditor_blocks#1]
![image](https://user-images.githubusercontent.com/3144571/152953018-39892e69-b43f-42ce-8f1c-54ef072128b8.png)

We got the icon to show up in the CKEditor toolbar, but it is not accessing the list of blocks. We are seeing these errors

This function no longer exists in Backdrop and should be unnecessary. 

`Error: Call to undefined function _block_rehash() in ckeditor_blocks_blocks_ajax() (line 70 of /app/modules/ckeditor_blocks/ckeditor_blocks.module).`

Not sure if this message is helpful yet.

`Notice: Undefined property: stdClass::$module in ckeditor_blocks_get_block() (line 41 of /app/modules/ckeditor_blocks/ckeditor_blocks.module).`

**Comment by stpaultim** [comment]
https://docs.backdropcms.org/api/backdrop/core%21modules%21layout%21layout.admin.inc/function/layout_block_list/1

https://docs.backdropcms.org/api/backdrop/core%21modules%21layout%21layout.module/function/layout_get_block_info/1

https://docs.backdropcms.org/api/backdrop/core%21modules%21layout%21layout.admin.inc/function/layout_block_add_page/1

**Comment by stpaultim** [comment]
I've confirmed that `$blocks = layout_get_block_info();` does get me a list of available blocks. 
![image](https://user-images.githubusercontent.com/3144571/152968053-4677ec69-9f8b-4a10-9252-c0ff579b772a.png)


The next step is to figure out how to make them available to CKEDITOR.

**Comment by laryn** [comment]
There’s a module that makes blocks available as a field — might provide some inspirations?

**Comment by stpaultim** [comment]
Good work @rajpaul - I merged your PR. 

This absolutely accomplishes the issue listed in this task. It accesses and displays a list of available blocks. 

With this PR, I was able to create a custom block through the custom block menu item and place it in a node using CKEditor. 

However, it did not work for a custom block created through the layout system and I was not able to use any existing blocks. 

But, I will open new tickets to address these issues.
